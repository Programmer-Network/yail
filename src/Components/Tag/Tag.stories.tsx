import { action } from "storybook/actions";

import Tag from "./index";

export default {
  title: "Components / Tag",
  parameters: {
    layout: "centered"
  }
};

export const Default = () => (
  <Tag
    name='Javascript'
    onClick={() => {
      action("onClick")("Javascript");
    }}
  />
);

export const WithoutClickHanddler = () => <Tag name='Javascript' />;
