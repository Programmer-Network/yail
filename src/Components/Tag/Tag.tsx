import classNames from "classnames";
import { FC } from "react";

import { ITagProps } from "./types";

const Tag: FC<ITagProps> = ({ name, className = null, onClick }) => {
  return (
    <div
      onClick={onClick}
      className={classNames(
        "text-primary hover:border-border hover:text-text rounded-md bg-transparent text-base last:mr-0",
        className,
        {
          "cursor-pointer": onClick
        }
      )}
    >
      <span>#{name}</span>
    </div>
  );
};

export default Tag;
