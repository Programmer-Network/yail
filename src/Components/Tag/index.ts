export { Tag } from "./Tag";
export * from "./types";
