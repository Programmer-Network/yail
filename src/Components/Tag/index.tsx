import classNames from "classnames";
import { FC } from "react";

import { ITagProps } from "./types";

const Tag: FC<ITagProps> = ({ name, className = null, onClick }) => {
  return (
    <div
      onClick={onClick}
      className={classNames(
        "yl-rounded-md yl-bg-transparent yl-text-base yl-text-primary last:yl-mr-0 hover:yl-border-border hover:yl-text-text",
        className,
        {
          "yl-cursor-pointer": onClick
        }
      )}
    >
      <span>#{name}</span>
    </div>
  );
};

export default Tag;
