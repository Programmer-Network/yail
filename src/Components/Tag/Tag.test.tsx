import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";

import Tag from "./";

describe("Tag component", () => {
  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(<Tag name='example' />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders tag with provided name", () => {
    const tagName = "example";
    render(<Tag name={tagName} />);
    expect(screen.getByText(`#${tagName}`)).toBeInTheDocument();
  });

  test("applies custom class name", () => {
    const customClassName = "custom-class";
    render(<Tag name='example' className={customClassName} />);
    const tag = screen.getByText("#example").parentNode;
    expect(tag).toHaveClass(customClassName);
  });

  test("handles click events", () => {
    const handleClick = vi.fn();
    render(<Tag name='example' onClick={handleClick} />);
    fireEvent.click(screen.getByText("#example"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("should have cursor-pointer class if onClick is provided", () => {
    const customClassName = "cursor-pointer";
    render(<Tag name='example' onClick={() => null} />);
    const tag = screen.getByText("#example").parentNode;
    expect(tag).toHaveClass(customClassName);
  });
});
