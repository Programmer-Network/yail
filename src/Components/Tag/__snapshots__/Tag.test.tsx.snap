// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <div
    class="yl-rounded-md yl-bg-transparent yl-text-base yl-text-primary last:yl-mr-0 hover:yl-border-border hover:yl-text-text"
  >
    <span>
      #example
    </span>
  </div>
</DocumentFragment>
`;
